lib/ziti-tunnel/tunnel_tcp.c

```diff
@@ -339,12 +346,25 @@ u8_t recv_tcp(void *tnlr_ctx_arg, struct raw_pcb *pcb, struct pbuf *p, const ip_
     struct tcp_hdr *tcphdr = (struct tcp_hdr *)((char*)p->payload + iphdr_hlen);
     u16_t src_p = lwip_ntohs(tcphdr->src);
     u16_t dst_p = lwip_ntohs(tcphdr->dest);
+    char src_str[IPADDR_STRLEN_MAX];
     char dst_str[IPADDR_STRLEN_MAX];
+    ipaddr_ntoa_r(&src, src_str, sizeof(src_str));
     ipaddr_ntoa_r(&dst, dst_str, sizeof(dst_str));
-    TNL_LOG(TRACE, "received segment %s:%d->%s:%d",
-            ipaddr_ntoa(&src), src_p, dst_str, dst_p);
-
     u8_t flags = TCPH_FLAGS(tcphdr);
+    char flags_str[40] = { 0 };
+
+    if (flags & TCP_FIN) strcat(flags_str, "FIN,");
```

Comment:
this seem like it would be slow to do that for every packet even if it is not logged (when log level < TRACE)